source/MapDetailPanel.cpp

```diff
@@ -283,11 +309,18 @@ bool MapDetailPanel::Click(int x, int y, int clicks)
 		return true;
 	}
 
+	const System *const previous = selectedSystem;
 	// The click was not on an interface element, so check if it was on a system.
 	MapPanel::Click(x, y, clicks);
 	// If the system just changed, the selected planet is no longer valid.
 	if(selectedPlanet && !selectedPlanet->IsInSystem(selectedSystem))
 		selectedPlanet = nullptr;
+	// If the system just changed, the planet scroll in the system needs to be reset.
+	if(selectedSystem != previous)
```

Comment:
Instead of always `GeneratePlanetCards` everywhere, do the same thing MapPanel does: Cache the previous selected system as a data member and then in Step check if the selected system changed.

Reply:
But this is what is done here, isn't it? Just that I don't stock it as a member, so maybe it could cause a problem at another place, but the only one that comes to my mind is when the player presses tab in which case the system changes

Reply:
This is in `Click`. There are other places where you call GeneratePlanetCards which you can avoid by moving this into Step. If that makes sense.

Reply:
Oh, right, so only have it at one place and check if the system was changed, got it!